test/helpers/kubectl.go

```diff
@@ -420,9 +437,73 @@ func (kub *Kubectl) CiliumReport(namespace string, pod string, commands []string
 		fmt.Fprintln(wr, out.CombineOutput())
 	}
 	fmt.Fprint(wr, "StackTrace Ends\n")
+	kub.CiliumReportDump(namespace, pod)
 	return nil
 }
 
+//CiliumReportDump runs a variety of commands(reportCommands) and writes the results to
+//testResultsPath
+func (kub *Kubectl) CiliumReportDump(namespace string, pod string) {
+	reportEndpointCommands := map[string]string{
+		"cilium endpoint get %s":    "endpoint_get_%s.txt",
+		"cilium bpf policy list %s": "bpf_policy_list_%s.txt",
+	}
+
+	testPath, err := ReportDirectory()
+	if err != nil {
+		kub.logCxt.WithError(err).Errof("Cannot create test result path '%s'", testPath)
+		return
+	}
+
+	reportCmds := map[string]string{}
+	for cmd, logfile := range reportCommands {
+		command := fmt.Sprintf("kubectl exec -n %s %s -- %s", namespace, pod, cmd)
+		reportCmds[command] = logfile
+	}
+	reportMap(testPath, reportCmds, kub.Node)
+
+	for _, ep := range kub.CiliumEndpointsIDs(pod) {
+		for cmd, logfile := range reportEndpointCommands {
+			command := fmt.Sprintf(cmd, ep)
+			res := kub.Node.Exec(fmt.Sprintf(
+				"kubectl exec -n %s %s -- %s", namespace, pod, command))
+			err = ioutil.WriteFile(
+				fmt.Sprintf("%s/%s", testPath, fmt.Sprintf(logfile, ep)),
+				res.CombineOutput().Bytes(),
+				0777)
```

Comment:
The amount of hardcoding needs to be reduced here. `0777` permission is used throughout this change. Make it a constant, name it accordingly, and make sure all uses of `0777` in the tests use this constant.